examples/seismic/tti/wavesolver.py

```diff
@@ -27,33 +27,51 @@ def __init__(self, model, source, receiver,
         self.receiver = receiver
 
         self.time_order = time_order
-        self.space_order = space_order/2
-
-        # Time step can be \sqrt{3}=1.73 bigger with 4th order
+        self.space_order = space_order
         self.dt = self.model.critical_dt
-        if self.time_order == 4:
-            self.dt *= 1.73
 
         # Cache compiler options
         self._kwargs = kwargs
 
     @cached_property
-    def op_fwd(self):
+    def op_fwd_shifted(self):
         """Cached operator for forward runs with buffered wavefield"""
         return ForwardOperator(self.model, save=False, source=self.source,
                                receiver=self.receiver, time_order=self.time_order,
-                               space_order=self.space_order, **self._kwargs)
+                               space_order=self.space_order,
+                               kernel='shifted', **self._kwargs)
 
     @cached_property
-    def op_fwd_save(self):
+    def op_fwd_shifted_save(self):
         """Cached operator for forward runs with unrolled wavefield"""
         return ForwardOperator(self.model, save=True, source=self.source,
                                receiver=self.receiver, time_order=self.time_order,
-                               space_order=self.space_order, **self._kwargs)
+                               space_order=self.space_order,
+                               kernel='shifted', **self._kwargs)
+
+    @cached_property
+    def op_fwd_centered(self):
+        """Cached operator for forward runs with buffered wavefield"""
+        return ForwardOperator(self.model, save=False, source=self.source,
+                               receiver=self.receiver, time_order=self.time_order,
+                               space_order=self.space_order,
+                               kernel='centered', **self._kwargs)
+
+    @cached_property
+    def op_fwd_centered_save(self):
+        """Cached operator for forward runs with unrolled wavefield"""
+        return ForwardOperator(self.model, save=True, source=self.source,
+                               receiver=self.receiver, time_order=self.time_order,
+                               space_order=self.space_order,
+                               kernel='centered', **self._kwargs)
+
+    def operator(self, kernel, save):
+        func_name = 'op_fwd_' + '%s' % kernel + ('save' if save else '')
```

Comment:
Better:
``func_name = "op_fwd_%s%s" % (kernel, 'save' if save else '')``
But, as I said above, I'd rather have a single method to be called as
``self.fwd_op(kernel, save)``

Reply:
I think it is linked to cached/not_cahed properties

Reply:
>But, as I said above, I'd rather have a single method

No